api/app/Http/Controllers/ChildEntityController.php

```diff
@@ -52,16 +74,19 @@ public function getAll($id)
      */
     public function getOne($id, $childId)
     {
-        $model = $this->getParentModel($id);
+        $this->validateId($id, $this->getParentModel()->getKeyName(), $this->validateParentRequestRule);
+        try {
+            $model =  $this->getParentEntityById($id);
+        } catch (ModelNotFoundException $e) {
+            throw $this->notFoundException($this->getParentModel()->getKeyName());
+        }
 
-        $relation = $this->getRelation($model);
-        $childKey = $this->getChildKey($relation);
-        $this->validateId($childId, $childKey, $this->validateChildRequestRule);
+        $this->validateId($childId, $this->getChildModel()->getKeyName(), $this->validateChildRequestRule);
 
-        $relation->getBaseQuery()->whereIn($childKey, [$childId]);
-        $childModel = $relation->getResults()->first();
-        if (!$childModel) {
-            throw $this->notFoundException($childKey);
+        try{
+            $childModel = $this->getChildEntityByIdAndParent($childId, $model);
+        }catch (ModelNotFoundException $e){
```

Comment:
Again, as `getChildEntityByIdAndParent` is a controller function, it can throw the desired exception directly.